pkg/beholder/authenticator.go

```diff
@@ -0,0 +1,71 @@
+package beholder
+
+import (
+	"errors"
+	"fmt"
+	"maps"
+)
+
+type Authenticator struct {
+	headers   map[string]string
+	publicKey []byte
+}
+
+func NewAuthenticator(config Config) (*Authenticator, error) {
```

Comment:
Does this need to be exported?

Reply:
Addressed in 210eb67fecd000768569ca6f2c43ea18859269ce